mdtk/degradations.py

```diff
@@ -183,25 +224,29 @@ def time_shift(excerpt, min_shift=50, max_shift=np.inf):
         offset = onset + excerpt.note_df.loc[note_index, 'dur']
```

Comment:
Generally iterating over indices like this is very slow. Is there a reason not to use `excerpt.note_df.iterrows()`? Ultimately I think you can do this process in a vectorised fashion - you're just making a boolean array ultimately. Play with setting `onset` to `excerpt.note_df.onset`, `offset = onset + excerpt.note_df.dur`, and using pandas series `.apply()` methods to apply `max()` and `min()` to apply these methods to every element. You'll end up with series results each time of the same size, then in the end you just combine them to make your bool (which you can get indices from if you want).

Reply:
This comment applies to wherever you're iterating like this in the file. I reckon we should merge this one, and you should make another branch to fix it later. I think we will need to for speed.